internal/gcs/bridge.go

```diff
@@ -0,0 +1,356 @@
+package gcs
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/guid"
+	"github.com/sirupsen/logrus"
+)
+
+type requestMessage interface {
+	SetActivityID(id guid.GUID)
+}
+
+type responseMessage interface {
+	Err() error
+}
+
+// rpc represents an outstanding rpc request to the guest
+type rpc struct {
+	proc rpcProc
+	id   int64
+	req  requestMessage
+	resp responseMessage
+	err  error
+	ch   chan struct{}
+}
+
+// bridge represents a communcations bridge with the guest. It handles the
+// transport layer but (mostly) does not parse or construct the message payload.
+type bridge struct {
+	Timeout time.Duration
```

Comment:
What timeout is this?

Reply:
Oh its per operation timeout?

Reply:
Ah, can you add the same comment here. `Timeout is the timeout a `sync` RPC must respond within.`